dynamiqs/sparse.py

```diff
@@ -0,0 +1,202 @@
+import functools
+
+import jax
+import jax.numpy as jnp
+from jaxtyping import Array
+
+
+class SparseDIA:
+    def __init__(self, diags: Array, offsets: tuple[int]):
+        self.offsets = offsets
+        self.diags = diags
+
+
+    def to_dense(self) -> Array:
+        """Turn any set of diagonals & offsets in a full NxN matrix."""
+        N = self.diags.shape[1]
+        out = jnp.zeros((N, N))
+        for offset, diag in zip(self.offsets, self.diags):
+            start = max(0, offset)
+            end = min(N, N + offset)
+            out += jnp.diag(diag[start:end], k=offset)
+        return out
+
+    @functools.partial(jax.jit, static_argnums=(0,1))
+    def _matmul_dense(
+        self,
+        direction: str,
```

Comment:
This is binary, so it should be a boolean. Something like:
```suggestion
        left_matmul: bool,
```